### PR TITLE
IDBVersionChangeEvent.newVersion can be null

### PR DESCRIPTION
**Pull request by japgolly**

### CODE REVIEW

**Comment by japgolly** [comment]
Spec: https://www.w3.org/TR/IndexedDB/#events

File: src/main/scala/org/scalajs/dom/raw/Idb.scala [hunk 1]
```diff
@@ -123,7 +124,7 @@ class IDBObjectStore extends js.Object {
 }
 
 trait IDBVersionChangeEventInit extends EventInit {
-  var newVersion: js.UndefOr[Int] = js.undefined
+  var newVersion: js.UndefOr[Int | Null] = js.undefined
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/raw/Idb.scala:127]: The convention so far has been to use the boxed classes when we have primitive-or-null. So this should be
```suggestion
  var newVersion: js.UndefOr[Integer] = js.undefined
```
In a future where Scala has explicit nulls, that will change. But we're not there yet.

File: src/main/scala/org/scalajs/dom/raw/Idb.scala [hunk 2]
```diff
@@ -142,9 +143,11 @@ class IDBVersionChangeEvent(typeArg: String,
   /**
    * Returns the new version of the database.
    *
+   * This is null when the database is being deleted.
+   *
    * MDN
    */
-  def newVersion: Int = js.native
+  def newVersion: Int | Null = js.native
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/raw/Idb.scala:150]:
```suggestion
  def newVersion: Integer = js.native
```

**Review by sjrd — CHANGES_REQUESTED** [review]
Thanks for the PR.

File: src/main/scala/org/scalajs/dom/raw/Idb.scala (hunk 1 above)
Comment (japgolly) [reply src/main/scala/org/scalajs/dom/raw/Idb.scala]: Np. Updated.

**Review by armanbilge — APPROVED** [review]

**Pull request merged**